### PR TITLE
Add Xcode 9, iOS 11, tvOS 11, and macOS 10.13 support

### PR DESCRIPTION
**Pull request by ZevEisenberg**
Wheeeee.

Maybe shouldn't merge this until Xcode is close to release, or release as a prerelease so people can use it during the summer beta period.

### CODE REVIEW

**Comment by ZevEisenberg** [comment]
Turns out, some of the floating point stuff [never worked on 32-bit devices](https://bugs.swift.org/browse/SR-5174). We're probably going to simplify the floating point conversions.

File: Source/Priority.swift [hunk 1]
```diff
@@ -50,20 +50,20 @@ public enum Priority: ExpressibleByFloatLiteral, ExpressibleByIntegerLiteral, Eq
         }
     }
 
-    public init(floatLiteral value: LayoutPriority) {
-        self.init(value)
+    public init(floatLiteral value: Float) {
+        self.init(Int(value))
```
Comment (jvisenti) [inline Source/Priority.swift]: Do we really want to be converting a `Float` priority to `Int` here?

File: Source/Priority.swift [hunk 2]
```diff
@@ -50,20 +50,20 @@ public enum Priority: ExpressibleByFloatLiteral, ExpressibleByIntegerLiteral, Eq
...
     }
 
     public init(_ value: Int) {
-        self = .custom(LayoutPriority(value))
+        self = .custom(LayoutPriority(Float(value)))
     }
 
     public init<T: BinaryFloatingPoint>(_ value: T) {
-        self = .custom(LayoutPriority(value))
+        self = .custom(LayoutPriority(Float(value)))
```
Comment (jvisenti) [inline Source/Priority.swift]: Is the `Float(value)` necessary? Is `LayoutPriority` no longer an alias for `Float`?

**Comment by jvisenti** [comment]
@ZevEisenberg Circle failures?

**Pull request merged**